presto-common/src/main/java/com/facebook/presto/common/block/MapBlockBuilder.java

```diff
@@ -470,6 +471,15 @@ public BlockBuilder newBlockBuilderLike(BlockBuilderStatus blockBuilderStatus)
                 newNegativeOneFilledArray(newSize * HASH_MULTIPLIER));
     }
 
+    @Override
+    public void reserve(int positionCount)
+    {
+        assert positionCount > 0;
+
+        mapIsNull = ensureCapacity(mapIsNull, positionCount);
+        offsets = ensureCapacity(offsets, positionCount + 1);
+    }
```

Comment:
curious why we don't ```updateDataSize``` here? It is coz the primitive types take care of it for complex types?

Reply:
@sujay-jain There is no `updateDataSize` in the MapBlockBuilder and RowBlockBuilder. I guess there were some historic reasons why they were treated differently. My understanding is the introduction of `updateDataSize` was to relieve the costly calculation in `getRetainedSizeInBytes()` but I don't have first hand data showing the benefit. So I'm just following the existing ways for now.